### ISSUE TITLE
[Bug] Using SGlang for inference always outputs token id 0

### ISSUE THREAD

**Issue by Sniper970119** [sgl-project/sglang#19908]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

I used slime for MoE SFT, and the loss was normal during the training process. 
When convert the Megatron checkpoint to hf checkpoint and conducting inference verification using SGlang at the end of the training, it was found that the model was unable to output normally. 
The specific manifestation is that regardless of the input, only the token "index:0" is output. Corresponding to the Qwen vocabulary, it is "!". At the same time, the logp value is also approximately -11. (For this step, I also conducted a verification using Llamafactory. The stored checkpoints that do not involve Megatron conversion still have this issue). 
However, this problem does not occur when switching to vllm, and normal output can be achieved. 
Since we will continue to use the RL framework based on SGlang for subsequent training, which really troubles me. I hope sombody can take a look and help me out.

Furthermore, since my model is based on the non-standard structure of Qwen3MoE (which is not any of the types of Qwen available for open-source), and because the current work is still ongoing and it is not convenient to open-source the specific checkpoints, the config.json is as follows: 

```json
{
  "architectures": [
    "Qwen3MoeForCausalLM"
  ],
  "attention_bias": false,
  "attention_dropout": 0.0,
  "bos_token_id": 151643,
  "decoder_sparse_step": 1,
  "dtype": "float32",
  "eos_token_id": 151643,
  "head_dim": 128,
  "hidden_act": "silu",
  "hidden_size": 1024,
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "max_position_embeddings": 32768,
  "max_window_layers": 28,
  "mlp_only_layers": [],
  "model_type": "qwen3_moe",
  "moe_intermediate_size": 384,
  "norm_topk_prob": true,
  "num_attention_heads": 16,
  "num_experts": 232,
  "num_experts_per_tok": 8,
  "num_hidden_layers": 28,
  "num_key_value_heads": 8,
  "output_router_logits": false,
  "qkv_bias": false,
  "rms_norm_eps": 1e-06,
  "rope_scaling": null,
  "rope_theta": 1000000,
  "router_aux_loss_coef": 0.001,
  "sliding_window": null,
  "tie_word_embeddings": true,
  "transformers_version": "4.57.1",
  "use_cache": true,
  "use_qk_norm": true,
  "use_sliding_window": false,
  "vocab_size": 151936
}

```

And I have verified that, using the config.json for random initialization and using sglang for inference, the above situation still occurs.

sglang inference result:
<img width="1112" height="607" alt="Image" src="https://github.com/user-attachments/assets/65826089-a8d1-45e5-b3d1-31f9fca8894e" />

vllm inference result:

<img width="1115" height="612" alt="Image" src="https://github.com/user-attachments/assets/b921ee83-9efd-4667-bf26-a15769360bd5" />

### Reproduction

CUDA_VISIBLE_DEVICES=0,1,2,3 python3 -m sglang.launch_server --model-path /model_path --host 0.0.0.0 --port 40003 --tp 4 --chunked-prefill-size 8192 --mem-fraction-static 0.6

CUDA_VISIBLE_DEVICES=4,5,6,7 vllm serve /model_path --host 0.0.0.0 --port 40002 --tensor-parallel-size 4 --max-num-batched-tokens 8192 --gpu-memory-utilization 0.6

### Environment

I used both SGlang 0.5.2 and 0.5.9(latest) and it still have problems. vllm version is 0.8.4.

**Comment by xingsy97** [comment]
created a fix #20127 for this bug, cc @Sniper970119 @yhyang201
